test/Constraints/diagnostics.swift

```diff
@@ -281,7 +281,7 @@ func r18800223(_ i : Int) {
 
   
   var buttonTextColor: String?
-  _ = (buttonTextColor != nil) ? 42 : {$0}; // expected-error {{result values in '? :' expression have mismatching types 'Int' and '(_) -> _'}}
+  _ = (buttonTextColor != nil) ? 42 : {$0}; // expected-error {{type of expression is ambiguous without more context}}
```

Comment:
I have mixed feelings because while `(_) -> _` is pretty ugly it also shows *why* we can't unify with `42`. But I don't know how often that comes up in practice.